modules/codegen-plugin/src/sbt-test/codegen-plugin/defaults/build.sbt

```diff
@@ -5,5 +5,41 @@ lazy val root = (project in file("."))
     libraryDependencies ++= Seq(
       "com.disneystreaming.smithy4s" %% "smithy4s-core" % smithy4sVersion.value,
       "com.disneystreaming.smithy4s" %% "smithy4s-dynamic" % smithy4sVersion.value
-    )
+    ),
+    /**
+      * this test is sensitive to whitespaces
+      * that's why we trim the whole thing and
+      * we filter out lines that are empty after whitespace trimming
+      */
+    TaskKey[Unit]("checkSmithyBuild") := {
+      val content =
+        IO.readLines(baseDirectory.value / "smithy-build.json")
+          .filter(_.trim().nonEmpty)
+          .mkString("\n")
+          .trim()
+      val expected = """|{
+                        |    "version": "1.0",
+                        |    "imports": [
+                        |        "src/main/smithy"
+                        |    ],
+                        |    "maven": {
+                        |        "dependencies": [
+                        |            "com.disneystreaming.alloy:alloy-core:0.2.6"
```

Comment:
just realized that this test will be sensitive to dependency upgrade (right now alloy is bumped in 0.18 and after merged series/0.18 here, it fails :/ )

Reply:
TBH I wouldn't check the file against a text version of JSON in the first place. I wouldn't want this to break because e.g. jsoniter changes ordering of serialized fields, or if we change config.

Just checking whether the file contains relevant strings should do (e.g. the `com.disneystreaming.alloy:alloy-core` prefix, `src/main/smithy`, `1.0`), and maybe also verifying that it all parses as JSON.

If you want to be more strict I'd still compare JSON ASTs rather than strings, and then the version numbers could maybe be ignored